pyvista/plotting/plotting.py

```diff
@@ -3782,6 +3789,14 @@ def on_timer(iren, event_id):
             self._add_observer("KeyPressEvent", self.key_press_event)
             self.update_style()
 
+        if self.virtual_display:
+            try:
+                from pyvirtualdisplay import Display
+            except:
+                raise RuntimeError("Cannot import Display from pyvirtualdisplay.")
+            self.display = Display(visible=0, size=(1024, 768), color_depth=24)
+            self.display.start()
+
```

Comment:
We may actually want this in the `BasePlotter` rather than the `Plotter`. 

Also, do we need to create the virtual display every time the plotter is instantiated or just once?

Reply:
That is a good point. I am not familiar with the internal structure of pyvista, so could you teach me? What are the benefits of moving function to Basepotter?

Personally, I think it's a good idea to create the display for each instance. This eliminates the need to worry about conflicts with other plotters.

Reply:
> I am not familiar with the internal structure of pyvista, so please teach me. What are the benefits of moving function to BasePlotter ?

Ah, `BasePlotter` is where we set up all the core functionality for plotting/rendering (and this feature falls into that category). In essence, the `Plotter` class is only for one specific case where we want the Plotter in a `vtkRenderWindow`. The `BasePlotter` is used in downstream projects for other plotting classes that will need this off-screen functionality. For example the `BackgroundPlotter` in `pyvistaqt` inherits from `BasePlotter` rather than `Plotter`.

> I think it's a good idea to create the display for each instance. This eliminates the need to worry about conflicts with other plotters.

Hm. Okay. Then we need to also close the display on the `close` call of the plotter.

----

So: We need to start the display in `BasePlotter.__init__` and stop it in the `BasePlotter.close` per https://github.com/ponty/pyvirtualdisplay/tree/1.3.2#usage

------

Before proceeding, I'd like to hear @pyvista/developers thoughts on this. Should we have one single display instance or try to manage these via the init/close of the base plotter.

Reply:
> Should we have one single display instance or try to manage these via the init/close of the base plotter.

It seems like the fewer global states the better, assuming `pyvirtualdisplay` can handle having multiple ones open. In practice I don't imagine people opening and keeping open hundreds of these -- or at least doing so and expecting things to work properly.

So my vote would be to let each `BasePlotter` manage their own and see if it works. I'd probably also document that at some point this might change to a single instance, so people should probably not rely on each one having its own (or there being a single shared one). If people want that level of control, they should probably fire up `xvfb` themselves and set `DISPLAY`, etc.

Reply:
@banesullivan @larsoner Thank you for your explanation and opinion. Now I am +1 for add this to ``BasePlotter``. 
>Hm. Okay. Then we need to also close the display on the close call of the plotter.

I will add this too.